geopolars/src/geoseries.rs

```diff
@@ -193,6 +287,42 @@ impl GeoSeries for Series {
         Series::try_from(("result", Arc::new(result) as ArrayRef))
     }
 
+    fn haversine_length(&self) -> Result<Series> {
+        use geo::algorithm::haversine_length::HaversineLength;
+        let mut result = MutablePrimitiveArray::<f64>::with_capacity(self.len());
+
+        for geom in iter_geom(self) {
+            let length: f64 = match geom {
+                Geometry::Point(_) => Ok(0.0),
+                Geometry::Line(line) => Ok(line.haversine_length()),
+                Geometry::LineString(line_string) => Ok(line_string.haversine_length()),
+                Geometry::Polygon(polygon) => Ok(polygon.exterior().haversine_length()),
+                Geometry::MultiPoint(_) => Ok(0.0),
+                Geometry::MultiLineString(multi_line_string) => {
+                    Ok(multi_line_string.haversine_length())
+                }
+                Geometry::MultiPolygon(mutli_polygon) => Ok(mutli_polygon
+                    .iter()
+                    .map(|poly| poly.exterior().haversine_length())
+                    .sum()),
+                Geometry::GeometryCollection(_) => {
+                    Err(PolarsError::ComputeError(std::borrow::Cow::Borrowed(
+                        "Length methods are not implemented for geometry collection",
+                    )))
```

Comment:
I guess in the future to support geometry collections it'll be easiest to make helper functions that can recurse into themselves.

Reply:
I am not 100% sure how geopandas handles this, if at all. It would make sense to recurse through them but I think for now, let's just  leave them unimplemented.